CAIPs/caip-104.md

```diff
@@ -0,0 +1,229 @@
+---
+caip: 104
+title: Namespace Reference Purpose and Guidelines
+status: Active
+type: Meta
+author: Juan Caballero (@bumblefudge), Wayne Chang (@wyc)
+created: 2022-03-27
+---
+
+# What is a Namespace Reference?
+
+Previously, the specifics of blockchain ecosystems and namespaces were defined
+in single stand-alone CAIPs alongside cross-chain CAIPs. Now, as the system has
+matured and expanded, it has become clear that namespaces are more complex and
+multi-dimensional than can be efficiently governed over time in the current CAIP
+single-document structure; instead, a Namespace Reference is proposed that
+splits out the application of each cross-chain CAIP to a given namespace as a
+distinct document with its own ratification/supersession lifecycle.  This way,
+long-term specifications for low-level primitives like Chain IDs and addresses
+can be defined once and not be superseded by the addition or modification of
+higher-level specifications for RPC interfaces or multiple asset types.
+
+In practical terms, that means each blockchain ecosystem's namespace is defined
+by a folder containing one document per applicable CAIP.  At a high-level, the
+important facts about a namespace, its governance, and its nomenclature can be
+defined/introduced in a `README.md` file, and each CAIP can be defined in a
+`caip{X}.md` file, where X is the number of the ratified CAIP.
+
+Like CAIPs, each CAIP-reference and namespace-reference is a design document
+providing information to the community and/or describing an addressing scheme
+with the explicit aim of further cross-chain (and more importantly,
+cross-namespace) engineering. The references should provide a concise technical
+specification of each feature or variant, as well as a rationale requiring
+little namespace-specific context. The namespace reference should include
+canonical (and ideally long-lived) links to authoritative documents, both when
+relied upon by the specified behavior AND when namespace-specific context is
+required to understand the inputs to or assumptions of the behavior.  
+
+The reference author is responsible for building consensus within the community
+and documenting dissenting opinions or rejected alternatives.
+
+## Namespace Reference Rationale
+
+Cross-chain engineering is difficult and often requires one to know both sides
+of a cross-namespace/cross-ecosystem interaction deeply to avoid serious
+security, UX, and design problems. The primary function of defining a namespace
+reference is to map the cross-chain CAIPs against the specifics of a given
+ecosystem and namespace. For this reason, it is important to consider a reader
+with very little context and tease out any potential risk of unfounded
+assumptions about:
+- what a blockchain or DAG is,
+- how transactions get validated against targeted chains or states, 
+- how addressing and routing work across different systems,
+- how a node's state at any given time relates to finality and the authoritative chain,
```

Comment:
I would remove this section - feels like it calls for extra prosa. IMHO readers of this document should know what a blockchain or DAG is. IMHO these documents should really be as concise as possible and not explain such things.